content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md

```diff
@@ -0,0 +1,239 @@
+
+Test suites, methods, test cases, and risk assessments are all carried out using gray box testing. The user interface of a software program is tested via black box testing.
+
+The following are the unit testing techniques:
+
+- Statements included in the scope
+- Coverage of the decision-making process
+- Coverage of Different Branches
+- Coverage of the Situation
+- Coverage by a Finite State Machine
+
+### Benefits of Unit testing
+
+- **More rapid development:**
+When developers write unit tests, they spend less time debugging and, as a result, have better confidence in making code changes. Developers will be more confident of adding new features to the product and subsequently refactor the code, which would not be possible without unit testing because confidence to act might be a problem if they are unfamiliar with what might break.
+
+- **Improved Design:**
+When developers build unit tests, they focus on how the System will utilize their code throughout, leading to better design.
+
+- **A More Effective Feedback Mechanism:**
+Once all of the unit tests have been completed, the state of the System may be assessed. These tests allow developers to assess other aspects of the code, such as whether it is complete or still being worked on. While changes in the test environment may influence the code base, the feedback mechanism can help signal how the test environment is doing.
+
+- **Tool for Pre-Regression:**
+Once developers have run all of the unit tests, they will be able to do any substantial refactoring or reorganization of code with good simplicity. Developers may be confident that their modifications will not impact other modules or the rest of the System in this way.
+
+- **Cost-cutting:**
+If a bug is detected late in the development process, it can be exceedingly costly to fix. A well-designed set of unit tests can detect fundamental defects (which could have a significant influence on the System later) early in the development process, lowering the cost of future maintenance.
+
+### Creating a project
+
+1. When creating a new project, we will use Visual Studio and create a new project from the start menu.
+2. Once we've done that, we'll need to locate our.NET Core's C# console app project template and click the icon that appears there.
+3. Decide on a name for your endeavor and then get to operate. Either the.NET Core 3.1 or.NET 5 target framework will have to be selected.
+
+   As you work on the School project, the Program.cs file will be open in the code editor.
+
+   > If you are unable to access the Program.cs file directly, you may do so by double-clicking it in Solution Explorer.
+
+4. To use the 'CollegeFinance' class, add the following C# code to Program.cs:
+
+```C#
+using System;
+namespace CollegeFinanceNS
+{   
+  public class CollegeFinance
+  {
+    private readonly string n_studentName;

```

Comment:
The variable name is `n_studentName` here but `n_studentIdentity` is being used below.